File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/communication/command/equipment/GetGunsCommand.java
package BaseLibrary.communication.command.equipment;

import BaseLibrary.communication.protocol.AProtocol;

public class GetGunsCommand extends AEquipmentCommand {

	private static final String NAME = "GUNS";

	static {
		AProtocol.registerForDeserialize(NAME, GetGunsAnswerCommand.class);
	}

	@Override
	public String getCommandName() {
		return NAME;
	}
	
	public GetGunsCommand() {
		pending = false;
	}
}

File: C#-Server+Client/BaseLibrary/communication/command/equipment/GetGunsCommand.cs
﻿using System.Collections.Generic;

namespace BaseLibrary.communication.command.equipment {
    public class GetGunsCommand : AEquipmentCommand {

        private static readonly List<ISubCommandFactory> SUB_COMMAND_FACTORIES = new List<ISubCommandFactory>();

        public static int RegisterSubCommandFactory(ISubCommandFactory subCommandFactory) {
            int position = SUB_COMMAND_FACTORIES.Count;
            SUB_COMMAND_FACTORIES.Add(subCommandFactory);
            return position;
        }

        public GetGunsCommand() : base() { }
    }
}
